src/parquet/parquet.h

```diff
@@ -117,6 +117,22 @@ class InMemoryInputStream : public InputStream {
   int64_t offset_;
 };
 
+// A wrapper for InMemoryInputStream to manage the memory.
+class ScopedInMemoryInputStream : public InputStream {
+ public:
+  ScopedInMemoryInputStream(const int64_t len);
+  ~ScopedInMemoryInputStream();
+  uint8_t* data();
+  int64_t size();
+  virtual const uint8_t* Peek(int num_to_peek, int* num_bytes);
+  virtual const uint8_t* Read(int num_to_read, int* num_bytes);
+
+ private:
+  int64_t len_;
+  uint8_t* buffer_;
```

Comment:
+1 -- I'm still acclimatizing to modern C++ (and thus have made a couple `new` mistakes, e.g. https://github.com/wesm/parquet-cpp/commit/0714c473f5ea7eaa0f6c5c4055028948951f93b9) so will endeavor to follow these guidelines.